sdk/scheduler/src/main/java/org/apache/mesos/util/JavaHomeDetector.java

```diff
@@ -0,0 +1,45 @@
+package org.apache.mesos.util;
+
+import org.apache.mesos.offer.TaskUtils;
+
+import java.util.Map;
+
+/**
+ * Determines the location of JAVA_HOME for a given task.
+ */
+public class JavaHomeDetector {
+    public static final String JAVA_HOME = "JAVA_HOME";
+    public static final String DEFAULT_JAVA_HOME = "jre1.8.0_91";
+    public static final String DEFAULT_JAVA_URI =
```

Comment:
I think it does occur today, that's why `DefaultOfferRequirementProvider` falls back to these defaults.

I think we can certainly not fallback on the default JRE. Let me make those changes.